pallas-network/src/multiplexer.rs

```diff
@@ -198,85 +214,90 @@ pub enum Error {
 
     #[error("plexer failed to mux chunk")]
     PlexerMux,
+
+    #[error("failure to abort the plexer threads")]
+    AbortFailure,
 }
 
-pub struct SegmentBuffer(Bearer, Vec<u8>);
+type Egress = (
+    tokio::sync::broadcast::Sender<(Protocol, Payload)>,
+    tokio::sync::broadcast::Receiver<(Protocol, Payload)>,
+);
 
-impl SegmentBuffer {
-    pub fn new(bearer: Bearer) -> Self {
-        Self(bearer, Vec::with_capacity(BUFFER_LEN))
+pub struct Demuxer(BearerReadHalf, Egress);
+
+impl Demuxer {
+    pub fn new(bearer: BearerReadHalf) -> Self {
+        let egress = tokio::sync::broadcast::channel(10_000);
```

Comment:
Make `10_000` a constant and increase to `100_000`. On my fast machine with localhost connection, I was able to overflow when it was only 10k.